Source/WebCore/editing/TextIterator.cpp

```diff
@@ -743,28 +744,28 @@ bool TextIterator::handleReplacedElement()
     if (m_fullyClippedStack.top())
         return false;
 
-    auto& renderer = *m_node->renderer();
+    auto& renderer = *m_currentNode->renderer();
     if (renderer.style().visibility() != Visibility::Visible && !m_behaviors.contains(TextIteratorBehavior::IgnoresStyleVisibility))
         return false;
 
     if (m_lastTextNodeEndedWithCollapsedSpace) {
-        emitCharacter(' ', *m_lastTextNode->parentNode(), m_lastTextNode, 1, 1);
+        emitCharacter(' ', Ref { *m_lastTextNode->parentNode() }, RefPtr { m_lastTextNode.get() }.get(), 1, 1);
         return false;
     }
 
     if (m_behaviors.contains(TextIteratorBehavior::EntersTextControls) && is<RenderTextControl>(renderer)) {
         if (auto innerTextElement = downcast<RenderTextControl>(renderer).textFormControlElement().innerTextElement()) {
-            m_node = innerTextElement->containingShadowRoot();
-            pushFullyClippedState(m_fullyClippedStack, *m_node);
+            m_currentNode = innerTextElement->containingShadowRoot();
+            pushFullyClippedState(m_fullyClippedStack, *protectedCurrentNode());
```

Comment:
Seems like this probably takes ownership so it should take `Ref` and caller should be doing `releaseNonNull` or have a `protectedNonNullCurrentNode`.

Reply:
This doesn't take ownership, it pushes a boolean to the stack.